src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -260,6 +266,43 @@ public sealed partial class JsonNumber : System.Text.Json.JsonNode, System.IEqua
         [System.CLSCompliantAttribute(false)]
         public bool TryGetUInt64(out ulong value) { throw null; }
     }
+    public sealed partial class JsonObject : System.Text.Json.JsonNode, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, System.Text.Json.JsonNode>>, System.Collections.IEnumerable
```

Comment:
`JsonArray` is planned to implement `IList`, but we decided not to make `JsonObject` implement `IDictionary`, because it would be then possible to pass it for methods that accepts a dictionary, which we didn't find useful and desired. @bartonjs ?

Reply:
@kasiabulat are there any methods from IDictionary that we would be missing if we wanted to implement that? Are there any Sort methods in IDictionary?

Reply:
There are no extension methods accepting IDictionary / IList instead of IEnumerable. LINQ queries work faster with IList and that's why making JsonArray implement it seemed valuable. No queries work faster with IDictionary - that's also why we thought implementing it by JsonObject would be unnecessary.

To implement IDictionary we are lacking: 
* `IsReadOnly`
*  `Keys` (but we have `PropertyNames`),
* `Values` (but we have `PropertyValues`)
* `Clear()`
* `CopyTo(T[], Int32)`
* `Contains(T)`
* `ContainsKey(TKey)` (but we have `ContainsProperty`)
* `TryGetValue(TKey, TValue)` (but we have `(Try)GetPropertyValue`).

Reply:
FYI use explicit implementation for the dictionary APIs people will likely never use - https://softwareengineering.stackexchange.com/questions/136319/whats-the-difference-between-implementing-an-interface-explicitly-or-implicitly

Reply:
I think my point was that unless there were compelling extension methods or other specific use cases that it probably didn't need to implement the IDictionary interface.  e.g. that one should be able to say "It implements IDictionary and IReadOnlyDictionary because [reason]" where "[reason]" is something more concrete than "because it's logically a dictionary" or "because it already had all the right methods".

Same result, different presentation: If no one will ever use the fact that it's an I(ReadOnly)Dictionary then it didn't need to assert the interface.

Reply:
Amoungst other benefits, IDictionary allows a collection initializer to be used:

```cs
var o = new JsonObject
{
    ["Name"] = "James",
    ["Radness"] = 98,
}
```

Reply:
> Amoungst other benefits, IDictionary allows a collection initializer to be used:

That doesn't require IDictionary.  Just IEnumerable + suitable Add methods, which it already has :smile: